src/main/java/com/esri/samples/symbology/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,354 @@
+      listView.setCellFactory(c -> new SymbolLayerInfoListCell());
+      // add the change listener to rebuild the preview when a selection is made
+      listView.getSelectionModel().selectedItemProperty().addListener(changeListener);
+      // add an empty entry to the list view to allow selecting 'nothing', and make it selected by default
+      listView.getItems().add(null);
+      listView.getSelectionModel().select(0);
+    }
+
+    // add colors to the color selection list view. We require the 0xAARRGGBB format to color the symbols
+    colorSelectionListView.getItems().addAll(0xFFFFFF00, 0xFFFF0000, 0xFF00FF00, 0xFF0000FF);
+    // make the first item in the list view selected by default
+    colorSelectionListView.getSelectionModel().select(0);
+    // add the change listener to the color selection list view, making a selection update the preview image
+    colorSelectionListView.getSelectionModel().selectedItemProperty().addListener(changeListener);
+    // add the cell factory to the color selection list view
+    colorSelectionListView.setCellFactory(c -> new ColorListCell());
+  }
+
+  /**
+   * Loads the stylx file and searches for all symbols contained within. Puts the resulting symbols into the GUI
+   * based on their category (eyes, mouth, hat, face).
+   */
+  private void loadSymbolsFromStyleFile() {
+    // create a SymbolStyle with the .stylx file
+    emojiStyle = new SymbolStyle("./samples-data/stylx/emoji-mobile.stylx");
+    emojiStyle.loadAsync();
+
+    // wait for the symbol style to load
+    emojiStyle.addDoneLoadingListener(() -> {
+      if (emojiStyle.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+        new Alert(Alert.AlertType.ERROR, "Error: could not load .stylx file. Details: " + emojiStyle.getLoadError().getMessage()).show();
+        return;
+      }
+
+      // load the default search parameters
+      ListenableFuture<SymbolStyleSearchParameters> defaultSearchParametersFuture = emojiStyle.getDefaultSearchParametersAsync();
+      defaultSearchParametersFuture.addDoneListener(() -> {
+        try {
+          SymbolStyleSearchParameters defaultSearchParameters = defaultSearchParametersFuture.get();
+
+          // use the default parameters to perform the search, getting all the available symbols within the file
+          ListenableFuture<List<SymbolStyleSearchResult>> symbolStyleSearchResultFuture = emojiStyle.searchSymbolsAsync(defaultSearchParameters);
+          symbolStyleSearchResultFuture.addDoneListener(() -> {
+            try {
+
+              // loop through the results and add each item to a list view according to category
+              List<SymbolStyleSearchResult> symbolStyleSearchResults = symbolStyleSearchResultFuture.get();
+              symbolStyleSearchResults.forEach(symbolStyleSearchResult -> {
+
+                // add the SymbolStyleSearchResult object to the correct list for its category
+                switch (symbolStyleSearchResult.getCategory().toLowerCase()) {
```

Comment:
I took over this from the original design. I suppose in theory they don't have to be made lower case, if we know exactly how the categories are formatted in the stylx file. 
I'd say it's a way to sanitize the data? In this case, it would catch `HAT`, `Hat`, `HaT`, etc.
Not entirely necessary here as we know the file is set up neatly, but I think it's considered good practice?

Long story short, I removed it, since it's technically unnecessary here.

Reply:
I took over this from the original design. I suppose in theory they don't have to be made lower case, if we know exactly how the categories are formatted in the stylx file. 
I'd say it's a way to sanitize the data? In this case, it would catch `HAT`, `Hat`, `HaT`, etc.
Not entirely necessary here as we know the file is set up neatly, but I think it's considered good practice?

Long story short, I removed it, since it's technically unnecessary here.